guest/api/types.go

```diff
@@ -57,6 +59,12 @@ type PreFilterPlugin interface {
 	PreFilter(state CycleState, pod proto.Pod) (nodeNames []string, status *Status)
 }
 
+// PreFilterExtension is a WebAssembly implementation of framework.PrefFilterExtensions.
+type PreFilterExtensions interface {
```

Comment:
I think you are supposed to add `Plugin` in the body, to inherit from the `Plugin` interface. Like so:

```suggestion
type PreFilterExtensions interface {
	Plugin

```